### PR TITLE
Always pass on transform context to providers, even minimal ones

### PR DESCRIPTION
**Pull request by nyalldawson**
## Description

[Replace this with some text explaining the rationale and details about this pull request]

### CODE REVIEW

**Review by nirvn — APPROVED** [review]

File: src/core/vectortile/qgsvectortilelayer.cpp [hunk 1]
```diff
@@ -385,7 +385,8 @@ bool QgsVectorTileLayer::writeSymbology( QDomNode &node, QDomDocument &doc, QStr
 
 void QgsVectorTileLayer::setTransformContext( const QgsCoordinateTransformContext &transformContext )
 {
-  Q_UNUSED( transformContext )
+  if ( mDataProvider )
+    mDataProvider->setTransformContext( transformContext );
```
Comment (m-kuhn) [inline src/core/vectortile/qgsvectortilelayer.cpp:389]: What happens if the VT layer is invalid initially and only loaded later, won't that miss out on this setter?
I.e. should the context be saved in a member variable of the layer and applied to a dataprovider in loadDataSource() as well?

Comment (nyalldawson) [reply src/core/vectortile/qgsvectortilelayer.cpp:389]: good point!

Comment (nyalldawson) [reply src/core/vectortile/qgsvectortilelayer.cpp:391]: Fixed, with a bunch of related improvements

**Review by m-kuhn — APPROVED** [review]

File: src/core/vectortile/qgsvectortilelayer.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/core/vectortile/qgsvectortilelayer.cpp:391]: Thanks!

**Pull request merged**